src/null-safety/index.md

```diff
@@ -135,22 +133,22 @@ listed by the analyzer.
 
 ### Dart 3 backwards compatibility
 
-Packages and apps migrated to use null safety with Dart 2.12 or later will
-likely be backwards compatible with Dart 3. Specifically, for any package where
-the lower bound of the SDK constraint is 2.12.0 or higher, pub will allow
-resolution even when the upper bound is limited to versions below 3.0.0. For
-example, a package with the following constraint will be allowed to resolve with
-a Dart 3.x SDK:
+Packages and apps migrated to use null safety with Dart 2.12 or later
+should be backwards compatible with Dart 3.
+If a any package has a lower bound SDK constraint of 2.12.0 or later,
+`dart pub` resolves that package.
+This remains true when the upper bound falls to a version earlier than 3.0.0.
+For example, a Dart 3.x SDK resolves a package with the following constraint:
 
 ```yaml
 environment:
   sdk: '>=2.14.0 <3.0.0'
 ```
 
-This allows developers to use Dart 3 sound null safety with packages that have
-been migrated to 2.12 null safety without needing a second migration.
-Note that this only applies to code that
-doesn't depend on Dart 3 breaking changes:
+This allows developers to use Dart 3 sound null safety with packages
+that use 2.12 or later null safety without a second migration.
+
+Note: This doesn't apply to code that doesn't depend on Dart 3 breaking changes.
```

Comment:
I think it's the other way around. It doesn't apply to code that _does_ depend on breaking changes in Dart 3.

```suggestion
Note: This doesn't apply to code that is affected by
other breaking changes in Dart 3.
```